api/sports-grid/construct-puzzle.js

```diff
@@ -0,0 +1,46 @@
+const seedrandom = require('seedrandom');
+
+const PUZZLE_SIZE = 3;
+
+const shuffle = (array, seed=null) => {
+    // use today's date as default seed
+    const rng = seedrandom(seed || new Date().toJSON().slice(0, 10));
+
+    for (let i = array.length - 1; i > 0; i--) {
+        const j = Math.floor(rng() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+}
+
+const hasOverlap = (group1, group2) => {
+    const setGroup1 = new Set(group1);
+    const intersection = group2.filter(x => setGroup1.has(x));
+    return (new Set(intersection)).size > 0;
+}
+
+const constructPuzzle = (teamsJSON) => {
+    const teams = Object.keys(teamsJSON);
+    const shuffledTeams = shuffle(teams);
```

Comment:
Does this shuffled mutate teams or create a shuffled copy?

Reply:
i think it mutates it because i didn't even end up using this variable lol.